### PR TITLE
Increase typing coverage for postgres provider

### PR DESCRIPTION
**Pull request by kenjihiraoka**
This PR is about increasing typing coverage for the postgres provider 
For more details, see Issue #9708

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by mik-laj** [comment]
Some static checks are sad. Can you fix it?

File: airflow/providers/postgres/hooks/postgres.py [hunk 1]
```diff
@@ -58,7 +59,9 @@ def __init__(self, *args, **kwargs):
         self.connection = kwargs.pop("connection", None)
         self.conn = None
 
-    def _get_cursor(self, raw_cursor):
+    def _get_cursor(self, raw_cursor: str) -> Union[psycopg2.extras.DictCursor,
+                                                    psycopg2.extras.RealDictCursor,
+                                                    psycopg2.extras.NamedTupleCursor]:
```
Comment (thepabloaguilar) [inline airflow/providers/postgres/hooks/postgres.py]: Would be great a type alias to the return type here

File: airflow/providers/postgres/operators/postgres.py [hunk 2]
```diff
@@ -60,9 +60,9 @@ def __init__(
         self.autocommit = autocommit
         self.parameters = parameters
         self.database = database
-        self.hook = None
+        self.hook: Optional[PostgresHook] = None
 
-    def execute(self, context):
+    def execute(self, context: Dict[Any, Any]) -> None:
```
Comment (thepabloaguilar) [inline airflow/providers/postgres/operators/postgres.py]: Are you sure that `context` has the `Dict` type?

File: airflow/providers/postgres/hooks/postgres.py [hunk 3]
```diff
@@ -149,7 +152,7 @@ def _serialize_cell(cell, conn):
         """
         return cell
 
-    def get_iam_token(self, conn):
+    def get_iam_token(self, conn: psycopg2.extensions.connection) -> Tuple[Any, Any, Union[int, Any]]:
```
Comment (thepabloaguilar) [inline airflow/providers/postgres/hooks/postgres.py]: The return type here is too general, I think you can improve it more!

**Review by thepabloaguilar — CHANGES_REQUESTED** [review]
Avoid to use `Any` if you know that return type is important and you know or can find the correctly type to put in. In some cases is not important like in the example below, but in the majority part of the time it's:
```python
def foo(iterable: Iterable[Any]) -> Any:
    return iterable[-1]
```

File: airflow/providers/postgres/hooks/postgres.py (hunk 3 above)
Comment (kenjihiraoka) [reply airflow/providers/postgres/hooks/postgres.py]: Thanks man, I found the exactly type of return of each one variables

File: airflow/providers/postgres/operators/postgres.py (hunk 2 above)
Comment (kenjihiraoka) [reply airflow/providers/postgres/operators/postgres.py]: Yes, good point man!
I dind't find exactly what is this variable, could it be better put just `Any`?

File: airflow/providers/postgres/hooks/postgres.py (hunk 1 above)
Comment (kenjihiraoka) [reply airflow/providers/postgres/hooks/postgres.py]: Is it gonna be something like this:

```
    conn_name_attr = 'postgres_conn_id'
    default_conn_name = 'postgres_default'
    supports_autocommit = True
    return_cursor_type = Union[psycopg2.extras.DictCursor, 
                               psycopg2.extras.RealDictCursor, 
                               psycopg2.extras.NamedTupleCursor]
    def __init__():
    
    def _get_cursor(self, raw_cursor: str) -> return_cursor_type:
```
Does it make sense?

Comment (thepabloaguilar) [reply airflow/providers/postgres/hooks/postgres.py]: Create the type alias outside of the class:
```python
_CURSOR_TYPE = Union[
    psycopg2.extras.DictCursor,
    psycopg2.extras.RealDictCursor, 
    psycopg2.extras.NamedTupleCursor
]

class Example:
    def _get_cursor(self, raw_cursor: str) -> _CURSOR_TYPE:
        ...
```

File: airflow/providers/postgres/operators/postgres.py (hunk 2 above)
Comment (thepabloaguilar) [reply airflow/providers/postgres/operators/postgres.py]: Maybe yes

**Comment by kenjihiraoka** [comment]
@mik-laj Can you help me? I'm not understanding what I'm doing wrong here:
`Signature of "_serialize_cell" incompatible with supertype "DbApiHook"`

**Comment by mik-laj** [comment]
@kenjihiraoka Parameter ``conn`` in ``DbApiHook`` is set to a default value when it really should always be passed. The following change should fix this problem.
```
--- a/airflow/hooks/dbapi_hook.py
+++ b/airflow/hooks/dbapi_hook.py
@@ -307,7 +307,7 @@ class DbApiHook(BaseHook):
         self.log.info("Done loading. Loaded a total of %s rows", i)

     @staticmethod
-    def _serialize_cell(cell: Any, conn: Any = None) -> Any:  # pylint: disable=unused-argument
+    def _serialize_cell(cell, conn):  # pylint: disable=unused-argument
         """
         Returns the SQL literal of the cell as a string.
```

**Comment by mik-laj** [comment]
@kenjihiraoka  Do you need any help?

**Comment by kenjihiraoka** [comment]
Do I need to do something to unblock this error?

Screenshot from 2020-09-06 17-39-01

**Comment by mik-laj** [comment]
@kenjihiraoka Can you do a rebase? that should solve your problem.

**Comment by kenjihiraoka** [comment]
Sorry man, I'll fix that wrong rebase :sweat:

**Comment by kenjihiraoka** [comment]
@mik-laj could you check if it's all okay? I couldn't identify if my changes broke some tests at CI pipe

File: airflow/providers/postgres/hooks/postgres.py (hunk 3 above)
Comment (SalAlba) [reply airflow/providers/postgres/hooks/postgres.py]: @kenjihiraoka please can you tell me the exactly type of return of each one variables ?

**Pull request closed without merging**